test/gpu/native/promoted-initializer.chpl

```diff
@@ -0,0 +1,12 @@
+use GpuDiagnostics;
+
+startGpuDiagnostics();
+
+on here.gpus[0] {
+    var A = 2 * [1,2,3,4,5];
+    writeln(A);
+}
+
+stopGpuDiagnostics();
+
+writeln(+ reduce getGpuDiagnostics().kernel_launch);
```

Comment:
I think we want to avoid using the number of kernel launches in the `.good` files in principle.
Could you use `assertGpuDiags(kernel_launch=1);`